float64 alpha1
float64 beta1
int32 err1
float64 alpha2
float64 beta2
int32 err2
